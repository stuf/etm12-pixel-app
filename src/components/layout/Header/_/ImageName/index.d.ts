import { FunctionComponent } from 'react';
import { Property } from 'kefir';

export interface Props {
  name: Property<string, any>;
  editing: Property<boolean, any>;
}

export interface Component extends FunctionComponent<Props> {}
