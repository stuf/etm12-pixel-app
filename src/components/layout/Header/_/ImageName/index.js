/* eslint no-unused-vars: [1, {"varsIgnorePattern": "[K|T]"}] */
/**
 * @module ImageName
 * @namespace components.layout.Header._
 */
import * as React from 'karet';
import * as U from 'karet.util';
import * as R from 'kefir.ramda';

import * as T from './index.d';
import styles from './index.module.scss';

/**
 * @param {T.Props} props
 * @return {T.Component}
 */
function ImageName({ name, isEditing }) {
  const toggleFlag = U.doModify(isEditing, R.not);

  return (
    <div className={U.cns(styles.root, 'flex-horizontal', '-vertical-center')}>
      {U.ifElse(
        isEditing,
        <div className="is-editing">
          <U.Input value={name} />
        </div>,
        <span className={styles.nameLabel} onClick={toggleFlag}>
          {name}
        </span>,
      )}
    </div>
  );
}

export default ImageName;
