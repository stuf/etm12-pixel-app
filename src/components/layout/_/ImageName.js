/* eslint no-unused-vars: [1, {"varsIgnorePattern": "[K|T]"}] */
/**
 * @module ImageName
 * @namespace components.layout.Header._
 */
import * as React from 'karet';
import * as U from 'karet.util';
import * as R from 'kefir.ramda';

function ImageName({ name, isEditing }) {
  const toggleFlag = U.doModify(isEditing, R.not);

  return (
    <div className={U.cns('imageName', 'flex-horizontal', '-vertical-center')}>
      {U.ifElse(
        isEditing,
        <div className="imageName__nameDisplay is-editing">
          <U.Input value={name} />
        </div>,
        <span className="imageName__nameLabel" onClick={toggleFlag}>
          {name}
        </span>,
      )}
    </div>
  );
}

export default ImageName;
