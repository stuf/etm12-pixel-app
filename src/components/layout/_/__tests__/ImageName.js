import * as React from 'karet';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

import ImageName from 'components/layout/_/ImageName';

it('snapshot test', () => {
  const w = shallow(<ImageName />);
  expect(toJson(w)).toMatchSnapshot('with empty data');
});

it('displays correct element based on isEditing prop', () => {
  const w = shallow(
    <ImageName {...{ isEditing: false, name: 'Untitled 123' }} />,
  );
  expect(w.find('.imageName__nameLabel').text()).toBe('Untitled 123');

  const w2 = render(
    <ImageName {...{ isEditing: true, name: 'Untitled 123' }} />,
  );
  expect(w2.find('.imageName__nameLabel')).toHaveLength(0);
  expect(w2.find('input')).not.toHaveLength(0);
});
