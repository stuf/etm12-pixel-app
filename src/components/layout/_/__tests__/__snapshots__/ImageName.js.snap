// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: with empty data 1`] = `
<div
  className="imageName flex-horizontal -vertical-center"
>
  <span
    className="imageName__nameLabel"
    onClick={[Function]}
  />
</div>
`;
